[jjchee77] iP

jjchee77 opened the pull request:
# Duke Chatbot

Duke is a simplistic task manager for everyone.

It can help you:

- [x]  Be more organized (it's a task manager)
- [x]  Financially 💰 (it's <u>completely</u> **FREE**)
- [ ]  ~~Save the world~~

> [!IMPORTANT] 
> The earth is in danger, save the trees by using a digital task manager.

## Features
Duke provides support for the following tasks by default:

* Todos
* Deadlines
* Events

_Need more features?_ Fear not, simple create it yourself.

Thanks to its open-source and scalable nature, simply _**add one yourself**_!

The source code conforms to a set of [strict coding standards](https://se-education.org/guides/conventions/java/intermediate.html).

The entry point to Duke can be located in `src/main/java/duke/Duke.java` as follows:

```java
public static void main(String[] args) {
  Duke duke = new Duke(TASKS_FILE_PATH, ARG_DELIMITER);
  duke.run();
}
```

## Installation
_Get your hands dirty!_

Here are the steps to get started

1. Install [Java 11](https://www.oracle.com/java/technologies/downloads/#java11)
2. Download the latest duke-<version>.jar from [here](https://github.com/jjchee77/ip/releases)
3. Run it using `java -jar duke-<version>.jar`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jjchee77 committed:
Add initial skeleton version

jjchee77 committed:
Separate output to different methods

jjchee77 committed:
Add user commands echo

jjchee77 committed:
Add list feature

jjchee77 committed:
Add marking of tasks

jjchee77 committed:
Add event, todo, deadline

jjchee77 committed:
Add semi automated testing

jjchee77 committed:
Add error handling

jjchee77 committed:
Add delete task

jjchee77 committed:
Add checkstyles

jjchee77 committed:
Add save feature

jjchee77 committed:
Merge branch 'branch-Level-7'

jjchee77 committed:
Add date and time parsing

jjchee77 committed:
Merge branch 'branch-Level-8'

jjchee77 committed:
Add more OOP concepts

jjchee77 committed:
Merge branch 'A-MoreOOP'

jjchee77 committed:
Normalize exceptions

jjchee77 committed:
Add more subpackages

jjchee77 committed:
Merge branch 'add-gradle-support'

jjchee77 committed:
Update gradle build

jjchee77 committed:
Add more exceptions

jjchee77 committed:
Add junit tests

jjchee77 committed:
Update gradle build

jjchee77 committed:
Update gradle build

jjchee77 committed:
Add javadocs

jjchee77 committed:
Update javadocs

jjchee77 committed:
Add find command

jjchee77 committed:
Fix coding style

jjchee77 committed:
Merge branch 'A-JavaDoc'

jjchee77 committed:
Merge branch 'A-CodingStandard'

jjchee77 committed:
Merge branch 'Level-9'

jjchee77 committed:
Enhance arguments validation

jjchee77 committed:
Add commands enum

jjchee77 committed:
Fix typo

jjchee77 committed:
Add logical separation of code

danielleloh commented on src/main/java/duke/commands/CommandBye.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    /**
+     * Executes the "bye" command, which displays a farewell message to the user.
```
For Javadocs overridden methods, maybe you can consider using the `@inheritDoc` tag? The same for the rest of your overridden method Javadocs.

jjchee77 replied:
Thank you! I will amend accordingly :)

danielleloh commented on src/main/java/duke/commands/CommandDeadline.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        tasks.add(this.deadline);
```
I like how you separated each action executed in your methods which shows a clear logical ordering of steps :)

jjchee77 replied:
Thank you :)

danielleloh commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,61 @@
+     * @param isDone The completion status of the task.
+     * @throws EmptyDescriptionException If the description is empty.
+     */
+    public Todo(String description, boolean isDone) throws EmptyDescriptionException {
```
I like that all your boolean variables are named properly and clearly to show its functions :)

jjchee77 replied:
Thank you :)

danielleloh commented on src/main/java/duke/exceptions/tasks/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,20 @@
+ * indicating that a task description cannot be empty.
+ */
+public class EmptyDescriptionException extends DukeException {
+    private static final String errorMessage = "The description of a task cannot be empty.";
```
Most of your other naming conventions for private static final constants are named in all caps, so it may be better for you to change this to all caps as well for consistency? (i.e. `ERROR_MESSAGE`)

jjchee77 replied:
Thanks for pointing this out! I will amend accordingly :)

jjchee77 committed:
Add javafx gui

jjchee77 committed:
Merge branch 'branch-Level-10'

jjchee77 committed:
Add usage of inheritDoc tag

jjchee77 committed:
Update version number

jjchee77 committed:
Add assertion

jjchee77 committed:
Refactor parser class

Parser class: Extract method to check arguments length

The Parser class has several methods that checks if the
arguments supplied by the user satisfy a minimum length.

This is duplicated code that can be extracted into a single
method. This allows for additional validation should there
be any that is required in future.

jjchee77 committed:
Merge pull request #2 from jjchee77/branch-A-Assertions

Add assertion

jjchee77 committed:
Merge pull request #1 from jjchee77/branch-A-CodeQuality

Refactor parser class

jjchee77 committed:
Enable assertions in gradle

jjchee77 committed:
Add deadline tasks sorting

All tasks are currently appended to the end of the
task list by default, with no way for the user to sort them.

Deadline tasks possess an inherent trait, a deadline that
can be used to chronologically sort them.

The new command "sort-deadlines" sorts tasks of type Deadline
in place.

Since it is debatable whether or not it is logically appropriate
to sort todos and events, we avoid them completely and simply
re-arrange the deadline tasks in-place.

jjchee77 committed:
Merge branch 'Branch-C-Sort'

jjchee77 committed:
Rename chatbot to Arc

jjchee77 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/arc/commands/CommandSortDeadline.java
#	src/main/java/arc/parser/Parser.java

jjchee77 committed:
Add title bar

jjchee77 committed:
Move storage constants

jjchee77 committed:
Add initial help message

jjchee77 committed:
Add parser test case

jjchee77 committed:
Add user guide

jjchee77 committed:
Update release version

jjchee77 committed:
Update docs readme
